enhancements/sig-architecture/138-workload-completion/README.md

```diff
@@ -0,0 +1,363 @@
+spec:
+  workload: ...
+  manifestConfigs:
+    - resourceIdentifier:
+        resource: jobs
+        name: some-job
+        namespace: default
+      completionRules:
+        - type: WellKnownCompletions
+status:
+  resourceStatus:
+    manifests:
+      - resourceMeta:
+          resource: jobs
+          name: some-job
+          namespace: default
+          ...
+        statusCompletions:
+          - completionTime: nil
+
+Run a job once and delete all associated manifests on completion
+
+```yaml
+apiVersion: work.open-cluster-management.io/v1
+kind: ManifestWork
+metadata:
+  namespace: <target managed cluster>
+  name: example-job
+spec:
+  deleteOption:
+    ttlSecondsAfterFinished: 30
+  manifestConfigs:
+    - resourceIdentifier:
+        resource: jobs
+        namespace: default
+        name: some-job
+      completionRules:
+        - type: WellKnownCompletions
+  workload: ...
+```
+
+Identify custom completion states for any arbitrary resource
+
+```yaml
+apiVersion: work.open-cluster-management.io/v1
+kind: ManifestWork
+metadata:
+  namespace: <target managed cluster>
+  name: example-custom-resource
+spec:
+  workload: ...
+  manifestConfigs:
+    - resourceIdentifier:
+        resource: mycustomresources
+        namespace: default
+        name: some-custom-resource
+      completionRules:
+        - type: JSONPaths
+          jsonPathsOperator: Or
+          jsonPaths:
+           - path: ".status.conditions[?(@.type=='Complete')].status"
+             value: "True"
+           - path: ".status.conditions[?(@.type=='Failed')].status"
+             value: "True"
+```
+
+### Test Plan
+- test that jobs and pods that are completed with WellKnownCompletions and no ttl are no longer updated
+- test that a workload that is completed with a ttl will be deleted after the time has ellapsed
+- test that a workload with custom JSONPaths completion rules are evaluated correctly
+
+### Graduation Criteria
+N/A
+
+### Upgrade Strategy
+It will need upgrade on CRD of ManifestWork on hub cluster, and upgrade of work agent on managed cluster.
+
+When a user needs to use this feature with an existing `ManifestWork`, the user needs to update the `ManifestWork` to
+add `completionRules` to any of the manifests that are required to complete before the `ManifestWork` should be considered complete.
+
+### Version Skew Strategy
+- The new fields are optional, and if not set, the manifestwork will be handled the same as by previous versions
+- Older versions of the work agent will ignore the newly added fields
+
+## Alternatives
+
+- Instead of adding the `JsonPathsOperator` to support logical OR rules, we could wait for CEL support in order to handle these
+types of expressions. The downside is that WellKnownCompletions for Jobs would not be expressable as a standard rule until the CEL work is complete, but considering that feature is planned for v1.0.0 it could be worth it to keep JsonPaths simple and avoid cluterring the API.
```

Comment:
I actually like the alternative approach better.

Reply:
The more I think about it I do agree CEL is a much better fit for this. JSONPaths is nice and simple to use, but it's just not flexible enough for the task.

Reply:
Would you want to see support for JSONPaths as a simple option for users in addition to CEL, or do you think it should only support CEL (and WellKnown... variants)?

Reply:
I guess there's really no reason to have JSONPaths other than that kubernetes users may be more familiar with it. 

But the simple cases like traversing an object path can be written pretty much identically in CEL. And more complicated cases like filtering a list are much easier.

Reply:
I think JSONPaths looks simple, but its capability is limited to be use in real use case.

Reply:
Agreed. I was just more familiar with JSONPaths myself, but as I have been experimenting with CEL a bit I can immediately see it is a much better fit. Rremoved JSONPaths from the design.